src/server/lib/getRegistrationLocation.ts

```diff
@@ -11,20 +15,35 @@ import { RegistrationLocation } from '@/shared/model/RegistrationLocation';
  */
 export const getRegistrationLocation = (
 	req: Request,
-	cmpConsentedAll = false,
 ): RegistrationLocation | undefined => {
-	const country: CountryCode | undefined = req.cookies['GU_geo_country'];
-
-	// registrationLocation must never be captured without full cmp consent
-	if (!cmpConsentedAll) return undefined;
+	/**
+	 * Cypress Test START
+	 *
+	 * This code checks if we're running in Cypress
+	 */
+	const runningInCypress = process.env.RUNNING_IN_CYPRESS === 'true';
+	if (runningInCypress) {
+		const cypressMockStateCookie = req.cookies['cypress-mock-state'];
+		return countryCodeToRegistrationLocation(cypressMockStateCookie);
+	}
```

Comment:
I suggest making this explicitly check for the specific values that you use in the test, rather than a generic value, as i can forsee a scenario where in a test this could be difficult to debug when using a mock state cookie.

```suggestion
	const runningInCypress = process.env.RUNNING_IN_CYPRESS === 'true';
	if (runningInCypress) {
		const cypressMockStateCookie = req.cookies['cypress-mock-state'];
		
		// check if the cookie value is an expected value
		const validCode = ['FR', 'GB', 'US'].some(code => code === cypressMockStateCookie);
		
		// if it is then return the code
		if (validCode) {
			return countryCodeToRegistrationLocation(cypressMockStateCookie);
		}
		
		// otherwise let it fall through to the default check below
	}
```